### PR TITLE
Webshot handler for scenario comparison map

### PR DESCRIPTION
**Pull request by yulia-bel**
### Overview

Second part of task id49, adding webshot handler for comparison map

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Please explain how to test the PR: ID of a dataset, steps to reach the feature, etc._

### Feature relevant tickets

_Link to the related task manager tickets_

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

File: api/libs/webshot/src/webshot.dto.ts [hunk 1]
```diff
@@ -34,26 +30,16 @@ export class WebshotPngConfig extends WebshotConfig {
...
-  @ApiProperty()
-  reportOptions!: ReportOptions;
 }
 
-export class WebshotSFComparisonMapPdfConfig extends WebshotConfig {
-  @ApiPropertyOptional()
-  @IsOptional()
-  pdfOptions?: PDFOptions;
-
+export class WebshotPdfReportConfig extends WebshotBasicPdfConfig {
```
Comment (hotzevzl) [inline api/libs/webshot/src/webshot.dto.ts:39]: I miss multiple inheritance 😬 
But it's ok. I don't think it's worth to use mixins here just for these tiny data structures.

**Review by hotzevzl — APPROVED** [review]
Thanks @yulia-bel - no strong opinions on the new changes, though lots of comments inline 😅  - feel free to decide what to do 😃

File: api/apps/api/src/modules/projects/projects-proxy.controller.ts [hunk 2]
```diff
@@ -79,6 +90,6 @@ export class ProjectsProxyController {
       );
```
[the remark this answers is not in the document]
Comment (yulia-bel) [reply api/apps/api/src/modules/projects/projects-proxy.controller.ts]: Done! mapAclDomainToHttpError() had all the possible errors with relevant error messages, duplications removed as well

File: api/apps/api/test/utils/webshot.service-fake.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      stream.push(pdfBuffer);
+      stream.push(null);
+
+      return right(stream);
+    } catch (error) {
+      return left(unknownPdfWebshotError);
+    }
+  }
+
+  async getPublishedProjectsImage(
```
[the remark this answers is not in the document]
Comment (yulia-bel) [reply api/apps/api/test/utils/webshot.service-fake.ts:33]: Here as well,  I think the best way will be to use real WebshotService (the same way it is done for png at run), so I left a todo note to completely remove the fake service once the FE integration is done

File: api/libs/webshot/src/webshot.dto.ts (hunk 1 above)
Comment (yulia-bel) [reply api/libs/webshot/src/webshot.dto.ts]: fixed :)

**Comment by yulia-bel** [comment]
@hotzevzl thanks for the review, all suggestions made total sense, so I made changed - once checks pass, will merge it

File: api/libs/webshot/src/webshot.dto.ts (hunk 1 above)
Comment (hotzevzl) [reply api/libs/webshot/src/webshot.dto.ts]: this was genuinely a wish related to easier composition in TS.

but I also like the way you changed this to make things simpler to follow 😄

**Pull request merged**